src/Microsoft.DotNet.Darc/src/DarcLib/Models/Darc/DependencyFlowGraph.cs

```diff
@@ -127,20 +127,219 @@ public void RemoveEdge(DependencyFlowEdge edge)
+                    {
+                        DependencyFlowNode child = edge.From;
+
+                        if (!visitedNodes[node].Contains(child) && !nodesToVisit.Contains(child))
+                        {
+                            if (visitedNodes.ContainsKey(child))
+                            {
+                                visitedNodes[child].AddRange(visitedNodes[node]);
+                            }
+                            else
+                            {
+                                visitedNodes.Add(child, new List<DependencyFlowNode>(visitedNodes[node]));
+                            }
+                            
+                            nodesToVisit.Enqueue(child);
+                        }
+                    }
+
+                    node.CalculateLongestPathTime();
+                }
+            }
+        }
+
+        /// <summary>
+        ///     Determine and mark the absolute longest build path in the flow graph, based on the Best Case time.
+        /// </summary>
+        public void MarkLongestBuildPath()
+        {            
+            // Find the node with the worst best case time, we will treat it as the starting point and walk down the path
+            // from this node to a product node
+            DependencyFlowNode startNode = Nodes.Aggregate( (n1, n2) => n1.BestCasePathTime > n2.BestCasePathTime ? n1 : n2);
+            startNode.OnLongestBuildPath = true;
+            MarkLongestPath(startNode);
+        }
+
+        public void MarkLongestPath(DependencyFlowNode node)
+        {
+            // The edges we are interested in are those that haven't been marked as on the longest build path 
+            // and aren't back edges, both of which indicate a cycle
+            var edgesOfInterest = node.OutgoingEdges.Where(e => !e.OnLongestBuildPath && !e.BackEdge).ToList();
+
+            if (edgesOfInterest.Count > 0)
+            {
+                DependencyFlowEdge pathEdge = edgesOfInterest.Aggregate( (e1, e2) => e1.To.BestCasePathTime > e2.To.BestCasePathTime ? e1: e2);
+
+                // Mark the edge and the node as on the longest build path
+                pathEdge.OnLongestBuildPath = true;
+                pathEdge.To.OnLongestBuildPath = true;
+                MarkLongestPath(pathEdge.To);
+            }
+        }
+
+        public static async Task<DependencyFlowGraph> BuildAsync(
             List<DefaultChannel> defaultChannels,
-            List<Subscription> subscriptions)
+            List<Subscription> subscriptions,
+            IRemote barOnlyRemote,
+            int days)
         {
             // Dictionary of nodes. Key is the repo+branch
             Dictionary<string, DependencyFlowNode> nodes = new Dictionary<string, DependencyFlowNode>(
                 StringComparer.OrdinalIgnoreCase);
             List<DependencyFlowEdge> edges = new List<DependencyFlowEdge>();
 
+            var rand = new Random();
```

Comment:
Where's this used?

Reply:
*sigh* removing it It was when I was putting in fake data.